python-sdk/src/astro/sql/operators/dataframe.py

```diff
@@ -215,6 +225,48 @@ def _convert_column_capitalization_for_output(function_output, columns_names_cap
             )
         return function_output
 
+    def get_openlineage_facets(self, task_instance) -> OpenLineageFacets:  # skipcq: PYL-W0613
+        """
+        Collect the input, output, job and run facets for DataframeOperator
+        """
+        input_dataset: list[OpenlineageDataset] = []
+        output_dataset: list[OpenlineageDataset] = []
+
+        if self.output_table:
```

Comment:
Now, we do not emit temp table events always.  you will have to handle that case here https://github.com/astronomer/astro-sdk/pull/1156

Reply:
I feel the output table can be temp if I'm using dataframe operator in some intermediate task and I do not want to persist data for longer periods @utkarsharma2 any thoughts on this?

Reply:
@phanikumv Ya, output_table can be a temp table, I think a check for the temp table is needed to stay consistent.

Reply:
@pankajastro / @utkarsharma2 how do i check that?

Reply:
like this..? 

`if self.output_table.openlineage_emit_temp_table_event():` ?